src/util.ts

```diff
@@ -367,34 +371,30 @@ export function* flatMap<In, Out>(rng: Iterable<In>, fn: (item: In) => Iterable<
     }
 }
 
-export function makeDebuggerEnvironmentVars(env: EnvironmentVariables): DebuggerEnvironmentVariable[] {
-    const converted_env: DebuggerEnvironmentVariable[] = Object.entries(env).map(
-        ([key, value]) => ({
-            name: key,
-            value
-        })
-    );
-    return converted_env;
-}
-
-export function mergeEnvironment(...env: EnvironmentVariables[]): EnvironmentVariables {
-    return env.reduce((acc, vars) => {
-        if (process.platform === 'win32') {
-            // Env vars on windows are case insensitive, so we take the ones from
-            // active env and overwrite the ones in our current process env
-            const norm_vars = Object.getOwnPropertyNames(vars).reduce<EnvironmentVariables>((acc2, key: string) => {
-                acc2[normalizeEnvironmentVarname(key)] = vars[key];
-                return acc2;
-            }, {});
-            return { ...acc, ...norm_vars };
-        } else {
-            return { ...acc, ...vars };
+export function makeDebuggerEnvironmentVars(env?: EnvironmentVariablesUndefined): DebuggerEnvironmentVariable[] {
+    if (!env) {
+        return [];
+    }
+    const converted_env: DebuggerEnvironmentVariable[] = [];
+    for (const [key, value] of Object.entries(env)) {
+        if (value) {
```

Comment:
I think it should be an explicit check against null `value === null`.  We want to keep `0` and `""`